compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -5102,13 +5079,6 @@ object Types extends TypeUtils {
     private var myReduced: Type | Null = null
     private var reductionContext: util.MutableMap[Type, Type] | Null = null
 
-    override def tryNormalize(using Context): Type =
-      try
-        reduced.normalized
-      catch
-        case ex: Throwable =>
-          handleRecursive("normalizing", s"${scrutinee.show} match ..." , ex)
```

Comment:
Now that you've put back constant folding, we should probably keep the handleRecursive.  Or, even better, put one in tryCompiletimeConstantFold.